src/sync.rs

```diff
@@ -704,15 +704,18 @@ impl<T: Copy> Default for LockFreeFrozenVec<T> {
     /// any heap allocations until the first time data is pushed to it.
     fn default() -> Self {
         Self {
-            data: [Self::NULL; NUM_BUFFERS],
+            data: std::array::from_fn(|_| Self::null()),
```

Comment:
nit: we can just use `[Self::null(); NUM_BUFFERS]` here

Reply:
So, we can't use `[Self::null; NUM_BUFFERS]` because `AtomicPtr` doesn't impl `Copy`. The workaround is to define some `const` (i.e., what happened before clippy said anything). With the new const context syntax it's possible to do `[const { Self::null() }; NUM_BUFFERS]` instead, which is what I did in the second commit.

Reply:
In that case I'd just allow the lint? I think the const is fine, and `from_fn` isn't guaranteed const so I'd rather not use it

Reply:
Another alternative is to do

```rust

impl<T: Copy> LockFreeFrozenVec<T> {
    const fn null() -> [AtomicPtr<T>; NUM_BUFFERS] {
        [const { AtomicPtr::new(std::ptr::null_mut()) }; NUM_BUFFERS]
    }
}
```

And replace calls to `[Self::NULL; NUM_BUFFERS]` with `Self::null()`

Reply:
works for me!